db.go

```diff
@@ -303,7 +303,8 @@ func (db *DB) Items() *ItemIterator {
 	return &ItemIterator{db: db}
 }
 
-func (db *DB) sync() error {
+// Sync commits the contents of the database to the backing FileSystem; this is a noop for an in-memory database. It must only be called while the database is opened.
```

Comment:
`Sync` needs to be guarded with the `db.mu` mutex.

Reply:
I would leave the `sync` method as is and just add a new public method that locks/unlocks the mutex and calls the private `sync`.

Reply:
So, replace the internal db.Sync call and the mutex things by it with that method? Why not just call *that* sync?

Reply:
That will work!

Reply:
Working on it...

Reply:
Checks have passed!